src/frontend/src/handler/create_mv.rs

```diff
@@ -200,35 +236,75 @@ pub async fn handle_create_mv_bound(
-
+        let mut with_options = get_with_options(handler_args.clone());
         let resource_group = with_options.remove(&RESOURCE_GROUP_KEY.to_owned());
 
-        if resource_group.is_some()
-            && !context
-                .session_ctx()
-                .config()
-                .streaming_use_arrangement_backfill()
-        {
-            return Err(RwError::from(ProtocolError("The session config arrangement backfill must be enabled to use the resource_group option".to_owned())));
+        let is_serverless_backfill = with_options
+            .remove(&CLOUD_SERVERLESS_BACKFILL_ENABLED.to_owned())
+            .unwrap_or_default()
+            .parse::<bool>()
+            .unwrap_or(false);
+
+        if resource_group.is_some() && is_serverless_backfill {
+            return Err(RwError::from(InvalidInputSyntax(
+                "Please do not specify serverless backfilling and resource_group together"
+                    .to_owned(),
+            )));
         }
 
         if !with_options.is_empty() {
             // get other useful fields by `remove`, the logic here is to reject unknown options.
             return Err(RwError::from(ProtocolError(format!(
                 "unexpected options in WITH clause: {:?}",
-                context.with_options().keys()
+                with_options.keys()
             ))));
         }
 
+        let sbc_addr = match SESSION_MANAGER.get() {
+            Some(manager) => manager.env().sbc_address(),
+            None => "",
+        }
+        .to_owned();
+
+        if is_serverless_backfill && sbc_addr.is_empty() {
+            return Err(RwError::from(InvalidInputSyntax(
+                "Serverless Backfill is disabled on-premise. Use RisingWave cloud at https://cloud.risingwave.com/auth/signup to try this feature".to_owned(),
+            )));
+        }
+
+        let resource_group = if is_serverless_backfill {
+            match provision_serverless_backfill(sbc_addr).await {
+                Err(e) => {
+                    return Err(RwError::from(ProtocolError(format!(
+                        "failed to provision serverless backfill nodes: {}",
+                        e.as_report()
+                    ))));
+                }
+                Ok(val) => Some(val),
+            }
+        } else {
+            resource_group
+        };
+
+        let context = OptimizerContext::from_handler_args(handler_args);
         let has_order_by = !query.order.is_empty();
         if has_order_by {
             context.warn_to_user(r#"The ORDER BY clause in the CREATE MATERIALIZED VIEW statement does not guarantee that the rows selected out of this materialized view is returned in this order.
 It only indicates the physical clustering of the data, which may improve the performance of queries issued against this materialized view.
 "#.to_owned());
         }
 
+        tracing::debug!(resource_group = resource_group, "after calling SBC");
```

Comment:
`after calling SBC` is only valid after calling SBC. I think you could simply print a final result here, while having the value from RPC printed on RPC completion.

Reply:
Sounds good.